frontend/openchat-client/src/openchat.ts

```diff
@@ -2906,31 +2936,70 @@ export class OpenChat extends OpenChatAgentWorker {
                         trackEvent("replied_to_message");
                     }
                 } else {
-                    this.removeMessage(
-                        chatId,
-                        msg.messageId,
-                        this.user.userId,
-                        threadRootMessageIndex,
-                    );
-                    failedMessagesStore.add(context, retryEvent);
-                    this.dispatchEvent(new SendMessageFailed(!canRetry));
+                    if (resp.kind == "rules_not_accepted") {
+                        this.markChatRulesAcceptedLocally(false);
+                    }                        
+
+                    if (resp.kind == "community_rules_not_accepted") {
+                        this.markCommunityRulesAcceptedLocally(false);
+                    }                        
+
+                    this.onSendMessageFailure(
+                        chatId, 
+                        msg.messageId, 
+                        threadRootMessageIndex, 
+                        event, 
+                        canRetry);
                 }
             })
             .catch((err) => {
-                this.removeMessage(
-                    chatId,
-                    event.event.messageId,
-                    this.user.userId,
-                    threadRootMessageIndex,
-                );
-                failedMessagesStore.add(context, retryEvent);
-                this._logger.error("Exception sending message", err);
-                this.dispatchEvent(new SendMessageFailed(!canRetry));
+                this.onSendMessageFailure(
+                    chatId, 
+                    event.event.messageId, 
+                    threadRootMessageIndex, 
+                    event, 
+                    canRetry,
+                    err);
             });
     }
 
     private canRetryMessage(content: MessageContent): boolean {
-        return content.kind !== "poll_content";
+        return content.kind !== "poll_content" && content.kind !== "crypto_content";
+    }
+
+    rulesNeedAccepting(): boolean {
+        const chatRules = this._liveState.currentChatRules;
+        const chat = this._liveState.selectedChat;
+        if (chat === undefined || chatRules === undefined) {
+            return false;
+        }
+
+        const communityRules = this._liveState.currentCommunityRules;
+        const community = this._liveState.selectedCommunity;
+
+        console.log(
```

Comment:
console.debug would be less annoying. With some easily filterable prefix like RULES.